src/components/postgresConnect.ts

```diff
@@ -316,6 +316,82 @@ class PostgresDB {
       return false
     }
   }
+
+  public getTemplateStages = async (templateId: number) => {
+    const text = 'SELECT id, number, title FROM template_stage WHERE template_id = $1'
+    try {
+      const result = await this.query({ text, values: [templateId] })
+      return result.rows
+    } catch (err) {
+      console.log(err.message)
+      throw err
+    }
+  }
+
+  public getCurrentStageHistory = async (applicationId: number) => {
+    const text =
+      'SELECT id, stage_id FROM application_stage_history WHERE application_id = $1 and is_current = true'
+    try {
+      const result = await this.query({ text, values: [applicationId] })
+      return result.rows[0]
+    } catch (err) {
+      console.log(err.message)
+      throw err
+    }
+  }
+
+  public addNewStageHistory = async (applicationId: number, stageId: number) => {
+    // Note: switching is_current of previous stage_histories to False is done automatically by a Postgres trigger function
+    const text =
+      'INSERT into application_stage_history (application_id, stage_id, time_created) VALUES ($1, $2, CURRENT_TIMESTAMP) RETURNING id'
+    try {
+      const result = await this.query({ text, values: [applicationId, stageId] })
+      return result.rows[0].id
+    } catch (err) {
+      console.log(err.message)
+      throw err
+    }
+  }
+
+  public getCurrentStatusHistory = async (applicationId: number) => {
+    const text = `SELECT application_status_history.id,
+      application_stage_history_id, status FROM
+      application_status_history JOIN application_stage_history ON
+      application_stage_history_id = application_stage_history.id WHERE
+      application_id = $1 AND application_status_history.is_current = true;`
+    try {
+      const result = await this.query({ text, values: [applicationId] })
+      return result.rows[0]
+    } catch (err) {
+      console.log(err.message)
+      throw err
+    }
+  }
+
+  public getCurrentStatusFromStageHistoryId = async (stageHistoryId: number) => {
+    const text =
+      'SELECT id, status FROM application_status_history WHERE application_stage_history_id = $1 AND is_current = true'
+    try {
+      const result = await this.query({ text, values: [stageHistoryId] })
+      return result.rows[0]
+    } catch (err) {
+      console.log(err.message)
+      throw err
+    }
+  }
+
+  public addNewStatusHistory = async (stageHistoryId: number, status = 'Draft') => {
```

Comment:
```suggestion
  public addNewStatusHistory = async (stageHistoryId: number, status: ApplicationStatus = 'Draft') => {
```

And remember to import the type `ApplicationStatus` from the `graphql.ts`file. This would prevent using this method passing some status not included in the enum.

Reply:
I just tried this, and the problem is that the GraphQL enum is expecting "DRAFT" (in caps), whereas Postgres is expecting "Draft". It seems like in general it's quite difficult to use the auto-generated GraphQL types in other contexts which haven't undergone the inflections.

Reply:
> > > Note: The user information is NOT added to the database correctly, as the current input plugin is not sending the response as correctly structured JSON, but the createUserFromApplication Action is working fine when responses are manually input.
> > 
> > 
> > Do we need to create another issue to fix that?
> 
> I'll probably try and include it when I do the issue to mutate Response after user interaction.

It's done in #34 (Mostly, it only sets the `text` field, we need to see how to handle the other two once we make plugins that use them)